drivers/sensor/nuvoton_adc_cmp_npcx/adc_cmp_npcx.c

```diff
@@ -0,0 +1,237 @@
+		if (ret)
+			break;
+
+		/* Then set lower or higher threshold */
+		control.param = ADC_NPCX_THRESHOLD_PARAM_L_H;
+		control.val = attr == SENSOR_ATTR_UPPER_THRESH ?
+			ADC_NPCX_THRESHOLD_PARAM_L_H_HIGHER :
+			ADC_NPCX_THRESHOLD_PARAM_L_H_LOWER;
+		ret = adc_npcx_threshold_ctrl_set_param(config->adc,
+						config->th_sel, &control);
+		break;
+
+	case SENSOR_ATTR_ALERT:
+		control.val = val->val1;
+		ret = adc_npcx_threshold_ctrl_enable(config->adc,
+						config->th_sel, !!control.val);
+		break;
+	default:
+		ret = -ENOTSUP;
+	}
+	return ret;
+}
+
+int adc_cmp_npcx_attr_get(const struct device *dev,
+			  enum sensor_channel chan,
+			  enum sensor_attribute attr,
+			  struct sensor_value *val)
+{
+	return -ENOTSUP;
+}
+
+int adc_cmp_npcx_trigger_set(const struct device *dev,
+			     const struct sensor_trigger *trig,
+			     sensor_trigger_handler_t handler)
+{
+	const struct adc_cmp_npcx_config *const config = DRV_CONFIG(dev);
+	struct adc_cmp_npcx_data *data = DRV_DATA(dev);
+	struct adc_npcx_threshold_control_t control;
+
+	if (trig == NULL || handler == NULL)
+		return -EINVAL;
+
+	if (trig->type != SENSOR_TRIG_THRESHOLD ||
+	    trig->chan != SENSOR_CHAN_VOLTAGE)
+		return -ENOTSUP;
+
+	data->handler = handler;
+
+	control.param = ADC_NPCX_THRESHOLD_PARAM_WORK;
+	control.val = (uint32_t)&data->work;
+	return adc_npcx_threshold_ctrl_set_param(config->adc, config->th_sel,
+						&control);
+}
+
+int adc_cmp_npcx_channel_get(const struct device *dev,
+			     enum sensor_channel chan,
+			     struct sensor_value *val)
+{
+	const struct adc_cmp_npcx_config *const config = DRV_CONFIG(dev);
+
+	if (chan != SENSOR_CHAN_VOLTAGE)
+		return -ENOTSUP;
+
+	if (val == NULL)
+		return -EINVAL;
+
+	val->val1 = (uint32_t)&config->adc;
+	return 0;
+}
+
+static const struct sensor_driver_api adc_cmp_npcx_api = {
+	.attr_set = adc_cmp_npcx_attr_set,
+	.attr_get = adc_cmp_npcx_attr_get,
+	.trigger_set = adc_cmp_npcx_trigger_set,
+	.channel_get = adc_cmp_npcx_channel_get,
+};
+
+#define NPCX_ADC_CMP_INIT(inst)                                               \
+	static struct adc_cmp_npcx_data adc_cmp_npcx_data_##inst;             \
+	static const struct adc_cmp_npcx_config adc_cmp_npcx_config_##inst = {\
+		.adc = DEVICE_DT_GET(                                         \
+			DT_IO_CHANNELS_CTLR(DT_INST(inst, DT_DRV_COMPAT))),   \
+		.th_sel = inst,                                               \
+		.chnsel = DT_IO_CHANNELS_INPUT(DT_INST(inst, DT_DRV_COMPAT)), \
+		.thrval = (uint16_t)COND_CODE_1(DT_NODE_HAS_PROP(             \
+			DT_DRV_INST(inst), threshold),                        \
+			(DT_PROP(DT_DRV_INST(inst), threshold)),              \
+			ADC_CMP_NPCX_UNDEFINED),                              \
+		.comparison = (enum adc_cmp_npcx_comparison)COND_CODE_1(      \
+			DT_NODE_HAS_PROP(DT_DRV_INST(inst), comparison),      \
+			(DT_STRING_TOKEN(DT_DRV_INST(inst), comparison)),     \
+			ADC_CMP_NPCX_UNDEFINED),                              \
```

Comment:
I see that there is no `DT_INST_STRING_TOKEN_OR`, but feel free to add one to the devicetree.h API, along with `DT_STRING_TOKEN_OR`, and test cases, if you'd like to have a generic helper.

Reply:
ACK